inspirehep/dojson/hep/fields/bd1xx.py

```diff
@@ -24,76 +24,137 @@
+                result.append({
+                    'type': 'INSPIRE ID',
+                    'value': i_value,
+                })
+
+            j_values = force_force_list(value.get('j'))
+            for j_value in j_values:
+                if j_value.startswith('JACoW-') or j_value.startswith('JACOW-'):
+                    result.append({
+                        'type': 'JACOW',
+                        'value': j_value[6:],
+                    })
+                elif j_value.startswith('ORCID:') and len(j_value) > 6:
+                    # XXX: also handle records that only contain 'ORCID:'.
+                    result.append({
+                        'type': 'ORCID',
+                        'value': j_value[6:],
+                    })
+                elif ORCID.match(j_value):
+                    result.append({
+                        'type': 'ORCID',
+                        'value': j_value,
+                    })
+                elif j_value.startswith('CCID-'):
+                    result.append({
+                        'type': 'CCID',
+                        'value': j_value,
+                    })
+
+            w_values = force_force_list(value.get('w'))
+            for w_value in w_values:
+                result.append({
+                    'type': 'INSPIRE BAI',
+                    'value': w_value,
+                })
+
+            return result
+
+        def _get_record(value):
+            x_value = force_single_element(value.get('x'))
+            return get_record_ref(x_value, 'authors')
+
+        return {
+            'affiliations': _get_affiliations(value),
             'alternative_name': value.get('q'),
-            'inspire_id': inspire_id,
-            'inspire_bai': value.get('w'),
-            'orcid': value.get('j'),
-            'record': person_record,
+            'curated_relation': value.get('y') == '1',
             'email': value.get('m'),
-            'affiliations': affiliations,
-            'profile': {"__url__": create_profile_url(value.get('x'))},
-            'curated_relation': value.get('y', 0) == 1
+            'full_name': _get_full_name(value),
+            'ids': _get_ids(value),
+            'record': _get_record(value),
+            'role': value.get('e'),
         }
-        # HACK: This is to workaround broken records where multiple authors
-        # got meshed up together.
-        if isinstance(ret['full_name'], (list, tuple)):
-            import warnings
-            warnings.warn("Record with mashed-up author list! Taking first author: {}".format(value))
-            ret['full_name'] = ret['full_name'][0]
-        return ret
 
     authors = self.get('authors', [])
+    values = force_force_list(value)
 
-    if key.startswith('100'):
-        authors.insert(0, get_value(value[0]))
-    else:
-        for single_value in value:
-            authors.append(get_value(single_value))
+    for value in values:
+        if key.startswith('100'):
```

Comment:
If there's more than `100` (rare, but happens) this will insert the second before the first, which is probably incorrect. The previous code was discarding it, so I guess this is still an improvement : )